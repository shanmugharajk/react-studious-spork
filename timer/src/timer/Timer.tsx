import * as React from "react";
import "./timer.css";

import { useCurrentTime } from "./use-current-time";

export const Timer: React.FunctionComponent = () => {
  const time = useCurrentTime();

  return (
    <div className="Timer__container">
      <span className="Timer__text">{time}</span>
    </div>
  );
};
