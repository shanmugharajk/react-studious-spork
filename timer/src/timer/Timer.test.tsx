import { render, screen } from "@testing-library/react";

import { Timer } from "./Timer";

describe("Timer", () => {
  beforeAll(() => {
    jest.useFakeTimers("modern");
    jest.setSystemTime(new Date("Thu May 20 2021 18:22:39 GMT+0530").getTime());
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it("should render Timer component", () => {
    const { container } = render(<Timer />);

    expect(container).toMatchSnapshot();
  });
});
