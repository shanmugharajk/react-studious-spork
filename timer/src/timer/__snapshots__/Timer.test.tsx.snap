// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer should render Timer component 1`] = `
<div>
  <div
    class="Timer__container"
  >
    <span
      class="Timer__text"
    >
      06:22:39 PM
    </span>
  </div>
</div>
`;
